catalog/spatial/wfs/spatial-wfs-featuretransformer-xstream/src/main/java/org/codice/ddf/spatial/ogc/wfs/transformer/xstream/XStreamWfsFeatureTransformer.java

```diff
@@ -0,0 +1,173 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.spatial.ogc.wfs.transformer.xstream;
+
+import com.thoughtworks.xstream.XStream;
+import com.thoughtworks.xstream.XStreamException;
+import com.thoughtworks.xstream.io.xml.WstxDriver;
+import com.thoughtworks.xstream.security.NoTypePermission;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.MetacardType;
+import java.io.InputStream;
+import java.util.List;
+import java.util.Optional;
+import javax.xml.namespace.QName;
+import org.apache.commons.lang3.StringUtils;
+import org.codice.ddf.spatial.ogc.wfs.catalog.converter.FeatureConverter;
+import org.codice.ddf.spatial.ogc.wfs.catalog.converter.FeatureConverterFactory;
+import org.codice.ddf.spatial.ogc.wfs.catalog.converter.impl.GmlEnvelopeConverter;
+import org.codice.ddf.spatial.ogc.wfs.catalog.converter.impl.GmlGeometryConverter;
+import org.codice.ddf.spatial.ogc.wfs.catalog.mapper.MetacardMapper;
+import org.codice.ddf.spatial.ogc.wfs.catalog.message.api.FeatureTransformer;
+import org.codice.ddf.spatial.ogc.wfs.catalog.message.api.WfsMetadata;
+import org.codice.ddf.spatial.ogc.wfs.catalog.metacardtype.registry.api.WfsMetacardTypeRegistry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class XStreamWfsFeatureTransformer<T> implements FeatureTransformer<T> {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(XStreamWfsFeatureTransformer.class);
+
+  private List<FeatureConverterFactory> featureConverterFactories;
+
+  private List<MetacardMapper> metacardMappers;
+
+  protected WfsMetacardTypeRegistry metacardTypeRegistry;
+
+  @Override
+  public Optional<Metacard> apply(InputStream document, WfsMetadata metadata) {
+    Metacard metacard = null;
+    XStream xStream = createXStream(metadata);
+    try {
+      metacard = (Metacard) xStream.fromXML(document);
+    } catch (XStreamException e) {
+      LOGGER.trace("Failed to parse FeatureMember into Metacard", e);
+    }
+
+    return Optional.ofNullable(metacard);
+  }
+
+  private XStream createXStream(WfsMetadata<T> metadata) {
```

Comment:
It may not be necessary to create an XStream object for every call to apply : `Once the XStream instance has been created and configured, it may be shared across multiple threads allowing objects to be serialized/deserialized concurrently`. Recommend initializing it once.